### PR TITLE
AP-2860 Put extra check for user management dirty flag

### PR DESCRIPTION
**Pull request by dreamind**

### CODE REVIEW

File: Apromore-Custom-Plugins/User-Admin-Portal-Plugin/src/main/java/org/apromore/plugin/portal/useradmin/UserAdminController.java [hunk 1]
```diff
@@ -777,6 +777,7 @@ private void setSelectedUsers(Set<User> users) {
         }
         updateAssignedRoleModel(users);
         updateAssignedGroupModel(users);
+        isUserDetailDirty = false; // extra defensive, ensure dirty is not set by field's setValue
```
Comment (raboczi) [inline Apromore-Custom-Plugins/User-Admin-Portal-Plugin/src/main/java/org/apromore/plugin/portal/useradmin/UserAdminController.java]: Would it make sense to remove the duplicate `isUserDetailDirty = false;` on line 730?  It just seems odd to set it twice in the same method.  The convention seems to be to set the dirty flags at the end of methods that mutate or update the current user or group.

File: Apromore-Custom-Plugins/User-Admin-Portal-Plugin/src/main/java/org/apromore/plugin/portal/useradmin/UserAdminController.java [hunk 2]
```diff
@@ -808,6 +809,7 @@ private Group setSelectedGroup(Group group) {
         nonAssignedUserModel.setMultiple(true);
         nonAssignedUserList = new AssignedUserListbox(nonAssignedUserListbox, nonAssignedUserModel, "Users not in the group");
         selectedGroup = group;
+        isGroupDetailDirty = false; // extra defensive, ensure dirty is not set by field's setValue
```
Comment (raboczi) [inline Apromore-Custom-Plugins/User-Admin-Portal-Plugin/src/main/java/org/apromore/plugin/portal/useradmin/UserAdminController.java]: Similarly, would it make sense to remove the duplicate `isGroupDetailDirty = false;` on line 784?

File: Apromore-Custom-Plugins/User-Admin-Portal-Plugin/src/main/java/org/apromore/plugin/portal/useradmin/UserAdminController.java [hunk 3]
```diff
@@ -265,7 +265,7 @@ public void doAfterCompose(Window win) throws Exception {
         mainWindow = win;
         String userName = portalContext.getCurrentUser().getUsername();
         currentUser = securityService.getUserByName(userName);
-        selectedUser = currentUser;
+        selectedUser = null;
```
Comment (raboczi) [inline Apromore-Custom-Plugins/User-Admin-Portal-Plugin/src/main/java/org/apromore/plugin/portal/useradmin/UserAdminController.java:268]: While this change isn't harmful, it seems a little less convenient that the old default of showing someone their own account when they open the user management window.

**Review by raboczi — APPROVED** [review]
I don't see anything harmful to warrant not approving this merge, but I have a few comments/queries.

File: Apromore-Custom-Plugins/User-Admin-Portal-Plugin/src/main/java/org/apromore/plugin/portal/useradmin/UserAdminController.java (hunk 3 above)
Comment (dreamind) [reply Apromore-Custom-Plugins/User-Admin-Portal-Plugin/src/main/java/org/apromore/plugin/portal/useradmin/UserAdminController.java:268]: Actually somewhere after that, we have setSelectedUsers(null); I think for now I just want to make sure nothing is selected to limit potential causes if the issue remains.

**Pull request merged**